store/pruning/manager.go

```diff
@@ -74,28 +76,52 @@ func (m *Manager) HandleSnapshotHeight(height int64) {
 		return
 	}
 
+	if m.snapshotInterval == 0 || uint64(height)%m.snapshotInterval != 0 {
+		// ensure that we keep only heights that match the current snap interval
+		return
+	}
+
 	m.pruneSnapshotHeightsMx.Lock()
 	defer m.pruneSnapshotHeightsMx.Unlock()
 
 	m.logger.Debug("HandleSnapshotHeight", "height", height)
 	m.pruneSnapshotHeights = append(m.pruneSnapshotHeights, height)
 	sort.Slice(m.pruneSnapshotHeights, func(i, j int) bool { return m.pruneSnapshotHeights[i] < m.pruneSnapshotHeights[j] })
+
+	if m.initFromDB && len(m.pruneSnapshotHeights) > 2 { // received 2 fresh snap heights as base level
```

Comment:
I am waiting for 2 fresh snapshots to release the loaded height. We can wait for more but there is no guarantee that the first snapshot after restart finishes before.